File: harness-core/src/main/java/top/wcpe/mc/testkit/harness/McTestkitProtocol.java
package top.wcpe.mc.testkit.harness;

/**
 * mc-testkit 冻结控制协议的常量（docs/API.md §3.4），纯 JDK。
 *
 * <p>桩与机器人之间经聊天 / 插件消息通道约定消息，载荷走 {@code :} 后缀。
 * 集群 / 崩溃接管等场景的「到达确认」「触发崩溃」标记属 template / 消费方约定，不进冻结协议。
 */
public final class McTestkitProtocol {

    /** 桩通知机器人「已就绪，可开始驱动」：{@code E2E_READY:<scenario>}。 */
    public static final String READY = "E2E_READY";

    /** 压测机器人向桩上报汇总前缀：{@code E2E_STRESS_RESULT:<载荷>}。 */
    public static final String STRESS_RESULT_PREFIX = "E2E_STRESS_RESULT:";

    /** 触发机器人在购买中主动断线（中断恢复场景）。 */
    public static final String DISCONNECT_NOW = "E2E_DISCONNECT_NOW";

    /** 经插件消息 UI 通道驱动时下发的会话 token。 */
    public static final String UI_TOKEN = "E2E_UI_TOKEN";

    private McTestkitProtocol() {
    }

    /** 拼接「就绪信号」完整消息：{@code E2E_READY:<scenarioId>}。 */
    public static String readyMessage(String scenarioId) {
        return READY + ":" + scenarioId;
    }
}

File: src/main/kotlin/top/wcpe/mc/testkit/contract/McTestkitProtocol.kt
package top.wcpe.mc.testkit.contract

/**
 * 对外契约④：机器人↔桩控制协议消息名。
 *
 * 机器人与桩通过聊天 / 插件消息通道约定这些消息（沿用首个接入项目的既有约定）。
 * 消息**载荷**由 `:` 后缀承载，如 `E2E_READY:<scenario>`。
 */
object McTestkitControlProtocol {
    /** 桩通知机器人「已装备就绪，可开始驱动」：`E2E_READY:<scenario>`。 */
    const val READY = "E2E_READY"

    /** 压测机器人向桩上报汇总：`E2E_STRESS_RESULT:ok=<n>,err=<n>,...`。 */
    const val STRESS_RESULT = "E2E_STRESS_RESULT"

    /** 触发机器人在购买中主动断线（中断恢复场景）：`E2E_DISCONNECT_NOW:<...>`。 */
    const val DISCONNECT_NOW = "E2E_DISCONNECT_NOW"

    /** 经插件消息 UI 通道驱动时下发的会话 token：`E2E_UI_TOKEN:<uuid>`。 */
    const val UI_TOKEN = "E2E_UI_TOKEN"
}

/**
 * 对外契约④：测试结论真源——桩写出的结果文件约定。
 *
 * 编排 / 任务**只认结果文件**判 PASS/FAIL，不靠日志猜测（NFR 正确性优先、架构不变量真源）。
 */
object McTestkitResultFile {
    /** 结果状态键。 */
    const val STATUS_KEY = "status"

    /** 结果说明键（失败原因写这里）。 */
    const val MESSAGE_KEY = "message"

    /** 通过状态值。 */
    const val PASS = "PASS"

    /** 失败状态值。 */
    const val FAIL = "FAIL"

    /** 某场景的结果文件名：`<scenario>.properties`。 */
    fun fileName(scenario: String): String = "$scenario.properties"
}
